### PR TITLE
Disable make processed data private

### PR DESCRIPTION
**Pull request by josenavas**
Built on top of #1141 

If the user is missing some of the columns that are required for our system, we don't allow it to request making the data private, since that's not possible until the missing columns are added. The system shows a message with the missing columns.

Fixes partially #1128

### CODE REVIEW

**Comment by antgonza** [comment]
Should it fully solve https://github.com/biocore/qiita/issues/1128 before is merged? If not, why?

**Comment by josenavas** [comment]
I can work extend this and just make a single PR, I was thinking on just making multiple, smaller PR.

**Comment by antgonza** [comment]
If that's the plan, should this be a new branch?

**Comment by josenavas** [comment]
IMOO is not leaving the master in an inconsistent state, but I can create it's own branch. What do you prefer, a single PR with all of it or 3 small PR in its own branch?

**Comment by antgonza** [comment]
I prefer 1 single PR, if not tooo complex.

**Comment by josenavas** [comment]
@antgonza I've included everything in this PR.

This now fixes #1128

**Comment by antgonza** [comment]
Thanks! Could you add test for the newly added code in: remove_add_prep_template, display_template
& PrepTemplatePanel.render?

**Comment by josenavas** [comment]
@antgonza I've created an issue since that involves refactor more code in qiita_pet: https://github.com/biocore/qiita/issues/1145

**Comment by antgonza** [comment]
Yup, @josenavas and I chatted and we agreed that adding those tests are a full refactor. Thus, :+1:

File: qiita_db/metadata_template/test/test_prep_template.py [hunk 1]
```diff
@@ -32,7 +32,8 @@
                            get_count)
 from qiita_db.metadata_template.prep_template import PrepTemplate, PrepSample
 from qiita_db.metadata_template.sample_template import SampleTemplate, Sample
-from qiita_db.metadata_template.constants import PREP_TEMPLATE_COLUMNS
+from qiita_db.metadata_template.constants import (
```
Comment (adamrp) [inline qiita_db/metadata_template/test/test_prep_template.py]: these values are in the __init__.py file, so you can import directly from `qiita_db.metadata_template`

**Comment by adamrp** [comment]
Juts a minor comment, but :+1:

File: qiita_db/metadata_template/test/test_prep_template.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/metadata_template/test/test_prep_template.py]: You're right! Fixed!

**Comment by josenavas** [comment]
@adamrp any remaining comment?

**Comment by josenavas** [comment]
I have 2 :+1: @adamrp can you merge if your are ok with the code? Thanks!

**Comment by josenavas** [comment]
AARGGGGH!!! I though this PR was merged!!! This is pretty bad as my current PR modifies some of the pages modified here! Can I get this merged so I can try to solve the potential hell of merge conflicts and complete my work? I'm surprised that I don't have any merge conflict on this one...

I'll be working on this tomorrow morning and this is kind of blocking further development on my other issue. @antgonza or @adamrp can you merge?

**Comment by josenavas** [comment]
Thanks @ElDeveloper

**Pull request merged**